jscomp/runtime/js_typed_array.ml

```diff
@@ -373,3 +373,77 @@ module Float64Array = struct
 end
 module Float64_array = Float64Array
 [@ocaml.deprecated "use `Float64Array` instead"]
+
+module DataView = struct
+  (** The DataView view provides a low-level interface for reading and writing
+      multiple number types in an ArrayBuffer irrespective of the platform's endianness.
+
+    @see <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView> MDN
+  *)
+
+  type t
+
+  external make : array_buffer -> t = "DataView" [@@bs.new]
+  external fromBuffer : array_buffer -> t = "DataView" [@@bs.new]
+  external fromBufferOffset : array_buffer -> int -> t = "DataView" [@@bs.new]
+  external fromBufferRange : array_buffer -> offset:int -> length:int -> t = "DataView" [@@bs.new]
+
+  external buffer : t -> array_buffer = "" [@@bs.get]
+  external byteLength : t -> int = "" [@@bs.get]
+  external byteOffset : t -> int = "" [@@bs.get]
+
+  external getInt8 : t -> int -> int = "" [@@bs.send]
+  external getUint8 : t -> int -> int = "" [@@bs.send]
+
+  external getInt16 : t -> int -> littleEndian:Js.boolean -> int = "" [@@bs.send]
+  external getInt16BigEndian : t -> int -> (_ [@bs.as 0]) -> int = "" [@@bs.send]
```

Comment:
But are you sure it doesn't do a strict equality check? I don't know node and the conventions they use, but it's far from inconceivable that people will do strict checks on boolean post-"good parts". 

You could do this in the meantime:
```ml
external getInt16LittleEndian : t -> int -> Js.boolean -> int = "" [@@bs.send]
let getInt16LittleEndian : t -> int -> int = fun self i -> getInt16LittleEndian self i Js.true_
 ```

For the `BigEndian` form you can just leave the argument off. It's optional.